weave/trace_server/calls_query_builder.py

```diff
@@ -1353,7 +1255,150 @@ def _create_like_optimized_in_condition(
         like_condition = _create_like_condition(field, like_pattern, pb, table_alias)
         like_conditions.append(like_condition)
 
-    return "(" + " OR ".join(like_conditions) + ")"
+    or_sql = "(" + " OR ".join(like_conditions) + ")"
+    if _field_requires_null_check(field):
+        return f"({or_sql} OR {table_alias}.{field} IS NULL)"
+    return or_sql
+
+
+class OptimizationConditions(BaseModel):
+    str_filter_opt_sql: Optional[str] = None
+
+
+def process_query_to_optimization_sql(
```

Comment:
reasoning in the abstract here, it seems there are some conditions which are not processable (return None). let's call those `unprocessable_operand`. In that case, if you have a condition: `unprocessable_operand OR processable_operand`, then the predicate filter will filter only to values of `processable_operand`. But if the `unprocessable_operand` is TRUE for some filtered subset, then this would be overly restrictive. Does that make sense?

Reply:
That does make sense, hmmm. We could return True?

Reply:
I think we need a test like this. Basically the nested resolution allows ORs to be split up in such a way that value results get filtered out if one part of the OR is not processable. I think you have to say: If any part of an OR is not processable, the entire OR is not processable

Reply:
I'll add that test! See some red in CI, then fix it

Reply:
wait @tssweeney  I already did that on line 1307:

```
for op in operation.or_:
    result = process_operand(op)
    if result is None:
        return None  # If any operand can't be optimized, the whole OR can't be optimized <--------
    or_conditions.append(result)
```

Reply:
Ideally we would fallback to the normal processing fn if we ever can't optimize

Reply:
oh nice!, yup that is perfect